### ISSUE TITLE
TreeView control not refreshed after opening new policy source where filter is active

### ISSUE THREAD

**Issue by kerframil** [Fleex255/PolicyPlus#12]
At present, I have a particular set of polices that have been established by directly altering the registry. These changes were enacted by the O&O Shutup10 utility. In addition to this, there are a small number of polices in the local GPO, which I originally established by using gpedit.msc.

Upon a filter being activated, the TreeView control in the left pane is refreshed in accordance with the filtered results. However, if one then uses File > Open Policy Resources to alter the source(s), this control is not refreshed. To force it to refresh, I have to toggle off View > Only Filtered Polices, then toggle it on again.

For example, this is my GPO - filtered to show configured items:-

![policyplus1](https://user-images.githubusercontent.com/889778/33068179-60afdb68-cea8-11e7-92ac-871f524d82e9.PNG)

I switch to a different policy source:-

![policyplus2](https://user-images.githubusercontent.com/889778/33068203-6ee40830-cea8-11e7-8b5a-2e4646d21571.PNG)

I toggle the filter view off and on. Only now does the left pane contain the correct nodes:-

![policyplus3](https://user-images.githubusercontent.com/889778/33068222-7bd13a18-cea8-11e7-974a-9ea3aa155290.PNG)

In short, where the filtered view is enabled, I think that the application should refresh the left pane automatically after altering the policy source(s).

**Comment by Fleex255** [comment]
Fixed in https://github.com/Fleex255/PolicyPlus/commit/3ed5766da2882f038285b0edd39736854268e6d0; some similar improvements made in https://github.com/Fleex255/PolicyPlus/commit/f9b11eef8f536576a385971d6ed1d554e0205091. The Debug build has been updated with all these changes. Please let me know whether things are working as intended for you. Thanks for the report!

**Comment by kerframil** [comment]
Indeed, it now works as expected. Duly closing.
